bazaar/src/tests.rs

```diff
@@ -53,261 +51,225 @@ impl ExtBuilder {
-        // initialisation
-        let cid = register_test_community::<TestRuntime>(None, 2);
-        let alice = AccountId::from(AccountKeyring::Alice);
-        let alice_shop = ShopIdentifier::from("QmW6WLLhUPsosBcKebejveknjrSQjZjq5eYFVBRfugygTB");
-        // upload dummy store to blockchain
-        assert!(
-            EncointerBazaar::new_shop(Origin::signed(alice.clone()), cid, alice_shop.clone())
-                .is_ok()
-        );
-
-        // get shops from blockchain
-        let shops = EncointerBazaar::shop_registry(cid);
-        let alices_shops = EncointerBazaar::shops_owned(cid, alice.clone());
-        // assert that shop was added
-        assert!(shops.contains(&alice_shop));
-        assert!(alices_shops.contains(&alice_shop));
-        // assert that the shop is owned by alice
-        assert_eq!(EncointerBazaar::shop_owner(&cid, &alice_shop), alice);
+        let cid = create_cid();
+        BusinessRegistry::<TestRuntime>::insert(cid, alice(), BusinessData { url: url(), last_oid: 3 });
+
+        assert!(EncointerBazaar::update_business(Origin::signed(bob()), cid, url1()).is_err());
+        assert!(EncointerBazaar::update_business(Origin::signed(alice()), CommunityIdentifier::zero(), url1()).is_err());
+
+        assert_eq!(EncointerBazaar::business_registry(cid, alice()), BusinessData { url: url(), last_oid: 3 });
     });
 }
 
 #[test]
-fn create_new_shop_with_bad_cid_fails() {
+fn delete_business() {
     ExtBuilder::build().execute_with(|| {
-        // initialisation
-        let alice = AccountId::from(AccountKeyring::Alice);
-        let alice_shop = ShopIdentifier::from("QmW6WLLhUPsosBcKebejveknjrSQjZjq5eYFVBRfugygTB");
-        let cid = CommunityIdentifier::from(blake2_256(&(0, alice.clone()).encode())); // fails to register cid
-
-        // assert that upload fails
-        assert!(
-            EncointerBazaar::new_shop(Origin::signed(alice.clone()), cid, alice_shop.clone())
-                .is_err()
-        );
-
-        // get shops from blockchain
-        let shops = EncointerBazaar::shop_registry(cid);
-        let alices_shops = EncointerBazaar::shops_owned(cid, alice);
-
-        // assert that shop was not added
-        assert_eq!(shops.contains(&alice_shop), false);
-        assert_eq!(alices_shops.contains(&alice_shop), false);
+        let cid = create_cid();
+        BusinessRegistry::<TestRuntime>::insert(cid, alice(), BusinessData { url: url(), last_oid: 2 });
+        BusinessRegistry::<TestRuntime>::insert(cid, bob(), BusinessData { url: url1(), last_oid: 3 });
+
+        assert!(EncointerBazaar::delete_business(Origin::signed(alice()), cid).is_ok());
+
+        assert_eq!(EncointerBazaar::business_registry(cid, alice()), BusinessData::default());
+        assert_eq!(EncointerBazaar::business_registry(cid, bob()), BusinessData { url: url1(), last_oid: 3 });
     });
 }
 
 #[test]
-fn removal_of_shop_works() {
+fn delete_business_inexistent() {
```

Comment:
I would prefer if the test names read as a sentence and are as explicit as possible. Here I am not certain if it should fail or not.

I believe that would be better:  `deleting_inexistent_business_is_err` or `deleting_inexistent_business_is_noop`, what do you think about that?

This also applies to the other tests.